### DISCUSSION TITLE
IPDB end of life: retirement plan

### DISCUSSION THREAD

**Discussion by svinota** [svinota/pyroute2#788] [General]
IPDB final EOL date is 18. September 2022. After that the module will be completely dropped from the code.

…

As you may have noticed, IPDB is deprecated already for some time (ok, some years already).

The issue with the module is that it has many (really, not one) architectural mistakes that make it impossible to support complex routing and rules setting. Same mistakes lead to the performance and memory consumption issues.

So I decided to retire the module and replace with NDB — a module with completely different architecture, but providing IPDB-like API. The API is not the same, though, thus some code changes may be needed.

…

In the discussion below you may leave your objections or comments to the IPDB retirement.

**Comment by dulek** [comment]
@svinota: Is this date still up to date?

**Reply by svinota** [reply]
Yep. If you have any objections, ideas, etc — your opinion is welcome.

**Reply by dulek** [reply]
Nah, I just need to keep in mind the ETA we have to transition to NDB.

**Reply by svinota** [reply]
The `pyroute2.ipdb` package will stay in the repo, but will be frozen and excluded from the `pyroute2`:s `setup.cfg` file.

It will be still possible though to install it with `pip`.

**Reply by dulek** [reply]
Alright, that'll be our backup plan to keep CI running. Thanks for reassurance!

**Reply by svinota** [reply]
We will not let it go down anyway. Not from our side, at least :)
